### DISCUSSION TITLE
Pushing cached stages to a repository simultaneously instead of sequentially

### DISCUSSION THREAD

**Discussion by AlphaMalik** [docker/buildx#1048] [Q&A] [answered]
Hi

I am taking advantage of the buildx's ability to push cached layers to an external registry. This comes in handy when building the same application multilple times a day over a distributed set of distinct machines. I have noticed that sometimes the exporting cache portion takes upto 8 minutes. The layers are pushed to the external registry sequentially one after the other. Is it possible to have all the layers pushed to the external registry simultaneously in parallel instead of pushing them to the registry one by one.

Thanks

**Comment by tonistiigi** [comment, marked as answer]
Indeed looks like cache upload is sequential while regular push is parallel. Open issue for tracking this.
